### DISCUSSION TITLE
Filter data based on permissions

### DISCUSSION THREAD

**Discussion by luiscoutinh** [spatie/laravel-permission#2201] [Q&A]
An application has companies and news. Each company has many news.

1. How can I validate if the user can view at least one company? For example, if the user has the permission `companies.view`, `companies.view.*` , `companies.view.1` or `companies.view.1,2` (without specifying the id's)?
2. How can get all companies if the user has the permission `company.view` or `company.view.*` or get only the companies with id 1 or 2, if the user has the permissions `company.view.1` and `company.view.2` or `companies.view.1,2`?
3. How can I get all news based in company permissions? This means show all the news if the user has the permission `company.view` or `company.view.*` or only the news that belongs to the companies with id 1 or 2, if the user has the permissions `company.view.1` and `company.view.2` or `companies.view.1,2`?

**Comment by erikn69** [comment]
Maybe bouncer
